### PR TITLE
Rename uninstall old module

### PR DESCRIPTION
**Pull request by UziTech**
### Description of the Change

Remove old package if an installed package is renamed

### Benefits

Won't have to manually uninstall the old package if a package is renamed

### Possible Drawbacks

Could break someones [workflow](https://xkcd.com/1172/)

### Applicable Issues

#600

### CODE REVIEW

File: src/install.coffee [hunk 1]
```diff
@@ -355,6 +355,10 @@ class Install extends Command
             @installNode (error) -> next(error, packagePath)
         commands.push (packagePath, next) =>
           @installModule(options, pack, packagePath, next)
+        if installGlobally and packageName isnt pack.name
+          commands.push (newPack, next) => # remove package with old name
```
Comment (50Wliu) [inline src/install.coffee]: I would make this comment a bit clearer; something like "package was renamed; delete old package folder"

**Comment by 50Wliu** [comment]
I can't tell for sure, but it looks like Travis might be failing with this change?

**Comment by UziTech** [comment]
It seems to be failing on one of the delete specs which this shouldn't affect. When just running the install specs everything works.

File: src/install.coffee [hunk 2]
```diff
@@ -355,6 +355,10 @@ class Install extends Command
             @installNode (error) -> next(error, packagePath)
         commands.push (packagePath, next) =>
           @installModule(options, pack, packagePath, next)
+        if installGlobally and packageName isnt pack.name
+          commands.push (newPack, next) => # package was renamed; delete old package folder
+            fs.removeSync(path.join(@atomPackagesDirectory, packageName))
```
Comment (UziTech) [inline src/install.coffee]: I looked into it further and it looks to be having an issue with this line. More specifically the `fs.removeSync` although I can't seem to figure out why it is throwing an error.

Comment (50Wliu) [reply src/install.coffee]: You can probably wrap this in a try/catch and log the error.

Comment (UziTech) [reply src/install.coffee]: Tried that. It doesn't display an error. It won't even let jasmine finish testing all the specs.

Comment (UziTech) [reply src/install.coffee]: Finally figured it out. The test-module2 fixture was giving the wrong name so it was failing on other tests.

**Comment by UziTech** [comment]
@50Wliu Could we get a new release for this?

**Comment by damieng** [comment]
Cause of issue https://github.com/atom/settings-view/issues/1018

**Pull request merged**